### ISSUE TITLE
Response(body.fromStream(ZStream.fromHub(hub)) has a one element delay

### ISSUE THREAD

**Issue by searler** [zio/zio-http#2284]
**Describe the bug**

A streaming Response is created from a Hub.

A sequence of elements (1,2,3..) is published to the hub.
An element N is only seen after N+1 has been published.

ZIO-HTTP 3.0.0-RC2
ZIO 2.0.15
Scala 3.3.0

**To Reproduce**

Given the code below
Connect to the server ```curl -i http://localhost:8080/sse```

Comparing the console and curl outputs demonstrates the one element delay. 

```scala
mport zio.Console.print
import zio.http.*
import zio.stream.{ZSink, ZStream}
import zio.{Chunk, Hub, Promise, ZIO}

object Simple extends zio.ZIOAppDefault {
  def run = {
    import zio.durationInt
    for {
      hub <- Hub.unbounded[String]

      _ <- ZIO.foreach(Range(1,1000))(i => hub.offer(s"$i\n").delay(3.seconds)).fork

      _ <- ZStream.fromHub(hub).run(ZSink.foreach(print(_))).fork

        ui = Http.collect[Request] { case Method.GET -> Root / "sse" =>
         Response(body = Body.fromStream(ZStream.fromHub(hub)))
      }

      _ <- Server.serve(ui).provide(Server.default)
    } yield ()
  }
}
```

**Comment by BranislavLazic** [comment]
This seems to be because of how `NettyBodyWriter#write` handles `StreamBody`:
```scala
      case StreamBody(stream, _, _)           =>
        stream.chunks
          .runFoldZIO(Option.empty[Chunk[Byte]]) {
            case (Some(previous), current) =>
              NettyFutureExecutor.executed {
                ctx.writeAndFlush(new DefaultHttpContent(Unpooled.wrappedBuffer(previous.toArray)))
              } *>
                ZIO.succeed(Some(current))
            case (_, current)              =>
              ZIO.succeed(Some(current))
          }
          .flatMap { maybeLastChunk =>
            // last chunk is handled separately to avoid fiber interrupt before EMPTY_LAST_CONTENT is sent
            ZIO.attempt(
              maybeLastChunk.foreach { lastChunk =>
                ctx.write(new DefaultHttpContent(Unpooled.wrappedBuffer(lastChunk.toArray)))
              },
            ) *>
              NettyFutureExecutor.executed {
                ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT)
              }.as(true)
          }
```

The response will be chunked, and for some reason, a "previous" chunk is written by Netty.
P.S. The same issue exists for Queue.

**Comment by guersam** [comment]
Seems like caused by #2156 to fix #2155.

Fortunately there is a dedicated test case for the original problem, so it will be easier to give another shot to solve the problem differently.
